tests/Unit/Exceptions/ExceptionHandlerTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Tests\Unit\Controllers\Store;
+
+use App\Exceptions\Handler;
+use Illuminate\Container\Container;
+use Illuminate\Support\Facades\Config;
+use League\OAuth2\Server\Exception\OAuthServerException;
+use ReflectionClass;
+use ReflectionException;
+use Tests\StoreTestCase;
+
+class ExceptionHandlerTest extends StoreTestCase
+{
+
+    private $exceptions = [];
+
+    protected function setUp()
+    {
+        // Tests reach into the container, and get upset when run in bulk with other tests.
+        $this->createApplication();
+        $this->exceptions = [
+            new OAuthServerException('mymessage', 4, "exception"),
+        ];
+    }
+
+    /** @test */
+    public function itShouldNotStackTraceInProduction()
+    {
+        $handler = Container::getInstance()->make(Handler::class);
+        try {
+            foreach ($this->exceptions as $exception) {
+                Config::set('app.env', 'production');
+                $this->assertFalse(
+                    $this->invokeMethod($handler, 'shallWeStackTrace', [$exception])
+                );
+            }
+        } catch (ReflectionException $e) {
+            // Just catch it.
+        }
+    }
+
+    /** @test */
+    public function itShouldStackTraceNotInProduction()
+    {
+        $handler = Container::getInstance()->make(Handler::class);
+        try {
+            foreach ($this->exceptions as $exception) {
+                // hashed to be "anything other than 'production'
+                Config::set('app.env', md5('production'));
```

Comment:
OK - but probs doesn't need to be so extreme. :)

Reply:
You're right, could have been just `local`